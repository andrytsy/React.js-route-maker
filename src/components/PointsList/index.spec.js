import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

import ConnectedPointsList, { PointsList } from './index.js'

describe('Test component: PointsList --- Snapshot', () => {
    it('+++capturing snapshot', () => {
        const renderedValue = renderer.create(<PointsList points = {[]} />).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});

describe('Test component: PointsList',() => {
    const initialState = { points: [], pointName: '' }
    const mockStore = configureStore()
    let store, wrapper

    beforeEach(()=>{
        store = mockStore(initialState)
        wrapper = mount(
            <Provider store={store}>
                <ConnectedPointsList />
            </Provider>
        )
    })

    it('+++ render the connected component', () => {
        expect(wrapper.length).toEqual(1)
    })

    it('+++ check Prop matches with initialState', () => {
        expect(wrapper.find(PointsList).prop('points')).toEqual(initialState.points)
    })
})