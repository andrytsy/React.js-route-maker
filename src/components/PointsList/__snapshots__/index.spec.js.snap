// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component: PointsList --- Snapshot +++capturing snapshot 1`] = `
<div
  className="list-container"
>
  <div
    className="list-container__input-block input-group"
  >
    <input
      className="input-group__input"
      onChange={[Function]}
      onKeyUp={[Function]}
      value=""
    />
    <div
      className="input-group__btn"
      onClick={[Function]}
    />
  </div>
  <ul
    className="list-container__points points-group"
  />
</div>
`;
